skbio/io/format/genbank.py

```diff
@@ -562,8 +562,7 @@ def _parse_locus(lines):
                ' +(bp|aa|rc)'
                ' +(.*DNA|.*RNA)?'
                ' +(linear|circular)?'
-               ' +(PRI|ROD|MAM|VRT|INV|PLN|BCT|VRL|PHG|'
-               'SYN|UNA|EST|PAT|STS|GSS|HTG|HTC|ENV|CON)'
+               ' +(?!.*DNA|.*RNA)([A-Z]{3})'
```

Comment:
We need this update when you want to write a embl record in genbank format and then read your genbank file: see [embl](https://github.com/bunop/scikit-bio/blob/issue-1483/skbio/io/format/tests/data/embl_single_record#L1) and [genbank](https://github.com/bunop/scikit-bio/blob/issue-1483/skbio/io/format/tests/data/genbank_single_record#L1) for the same object `M14399`: the genbank division is `BCL` while the embl is `PRO`. If I write the embl version, I will write the `PRO` division in genbank locus (header), and then I cannot use the genbank parser since `PRO` is not specified in pattern. I could express all EMBL division in genbank (but I don't like it, what happens if embl change division? or I wrote a new parser for a different nucleotide database?), or read any tree letters as division. If you read any three letter as division, you could read also DNA as division, like in [this](https://github.com/bunop/scikit-bio/blob/issue-1483/skbio/io/format/tests/test_genbank.py#L236) test: this test need to fail (as we lack division) but the greedy re `([A-Z]{3})` catch DNA, while the two previous re could be empty: 
```python
import re
import pprint

# this is the example in test_genbank.test_parse_locus_invalid
lines = [
    # missing unit
    ['LOCUS       NC_005816               9609 '
     '    DNA     circular CON 07-FEB-2015'],
    # missing division
    ['LOCUS       SCU49845     5028 bp'
     '    DNA                    21-JUN-1999'],
    # wrong date format
    ['LOCUS       NP_001832                360 aa'
     '            linear   PRI 2001-12-18']]

# getting the missing division example
line = lines[1][0]

# define a generic 3 letters pattern
pattern = (
   r'LOCUS'
   ' +([^\s]+)'
   ' +([0-9]+)'
   ' +(bp|aa|rc)'
   ' +(.*DNA|.*RNA)?'
   ' +(linear|circular)?'
   ' +([A-Z]{3})'
   ' +([0-9]{2}-[A-Z]{3}-[0-9]{4})')

# search the pattern
matches = re.match(pattern, line[0])

# reading values like _parse_locus does
res = dict(zip(
    ['locus_name', 'size', 'unit', 'mol_type',
     'shape', 'division', 'date'],
    matches.groups()))

# getting values
pprint.pprint(res)
{'date': '21-JUN-1999',
 'division': 'DNA',
 'locus_name': 'SCU49845',
 'mol_type': None,
 'shape': None,
 'size': '5028',
 'unit': 'bp'}
```
This is the reason why I need to exclude DNA and RNA from division, in order to pass the `test_genbank.test_parse_locus_invalid`. Sorry for this confusion, but I think that all tests needed are already coded in `test_genbank` and `test_embl`